resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/providers/sse/SseEventInputImpl.java

```diff
@@ -30,16 +33,20 @@
    private volatile boolean isClosed = false;
 
    private boolean lastFieldWasData;
+   
+   private boolean escape = false;
 
    private final String DELIMITER = new String(SseConstants.EVENT_DELIMITER, StandardCharsets.UTF_8);
 
-   public SseEventInputImpl(Annotation[] annotations, MediaType mediaType, MultivaluedMap<String, String> httpHeaders,
-         InputStream inputStream)
+   public SseEventInputImpl(Annotation[] annotations, MediaType streamType, MediaType elementType,
+         MultivaluedMap<String, String> httpHeaders, InputStream inputStream)
    {
       this.annotations = annotations;
-      this.mediaType = mediaType;
+      this.mediaType = elementType;
       this.httpHeaders = httpHeaders;
       this.inputStream = inputStream;
+      this.textLike = MediaTypeHelper.isTextLike(streamType);
```

Comment:
Based on line 49, it seems that streamType can be null. What about to add something like `streamType != null && ` or `streamType == null || ` here (line 48) to protect NPE?